nix: upgrade from 2.6.1 to 2.8.0

jakubgs opened the pull request:
Minor upgrade done to match the version deployed in CI due to a bug with `nix-copy-closure` which broken Nix cache jobs:
https://github.com/status-im/infra-ci/issues/49

This will __not__ force upgrade on developes, nor require any `make nix-purge`. It is __optional__.
If however and upgrade is wanted it can be simply done with `nix upgrade-nix`.

jakubgs committed:
nix: upgrade from 2.6.1 to 2.8.0

Minor upgrade done to match the version deployed in CI due to a bug
with `nix-copy-closure` which broken Nix cache jobs:
https://github.com/status-im/infra-ci/issues/49

This will not force upgrade on developes, nor require any `make nix-purge`.
If however and upgrade is wanted it can be simply done with `nix upgrade-nix`.

Signed-off-by: Jakub Sokołowski <jakub@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 814af0f0 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-13325/1/) | 2022-05-05 19:27:22 | ~9 min | `ios` | [:package:`ipa`](https://i.diawi.com/PYnvGy) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FPYnvGy)|

jakubgs committed:
ci: use nix copy instead of nix-copy-closure

Also use `ssh-ng://` protocol implementation instead of `ssh://`.
Might fix the following error:
```
error (ignored): error: reached end of FramedSource
error: sized: unexpected end-of-file
error: unexpected end-of-file
```
For more details see: https://github.com/NixOS/nix/issues/6495

Signed-off-by: Jakub Sokołowski <jakub@status.im>
